### DISCUSSION TITLE
How to stop copier.copy() process

### DISCUSSION THREAD

**Discussion by readonlymem0ry** [pschatzmann/arduino-audio-tools#1544] [Q&A] [answered]
How to stop "copy()" process? (like player.copy(), copier.copy(), etc)

**Comment by pschatzmann** [comment, marked as answer]
- by stopping to call copy()
- or by setting the copier or the audio player to inactive e.g. with setActive(false)

**Reply by readonlymem0ry** [reply]
How do I know if the audio playing has reached its end or stopped?

**Reply by pschatzmann** [reply]
You are starting to embarrass yourself: did you look at the documentation of the copy method ?

**Reply by readonlymem0ry** [reply]
I'm just asking. thank you
